main.go

```diff
@@ -21,68 +20,69 @@ func main() {
 		&cli.StringFlag{
 			Name:    "surge-token",
 			Usage:   "token for surge authentication",
-			EnvVars: []string{"PLUGIN_SURGE_TOKEN"},
+			Sources: cli.EnvVars("PLUGIN_SURGE_TOKEN"),
 		},
 		&cli.StringFlag{
 			Name:    "path",
 			Usage:   "path to upload files from",
-			EnvVars: []string{"PLUGIN_PATH"},
+			Sources: cli.EnvVars("PLUGIN_PATH"),
 		},
 		&cli.StringFlag{
 			Name:    "pipeline-event",
 			Usage:   "event of the current pipeline",
-			EnvVars: []string{"CI_PIPELINE_EVENT", "CI_BUILD_EVENT"},
+			Sources: cli.EnvVars("CI_PIPELINE_EVENT", "CI_BUILD_EVENT"),
 		},
 		&cli.StringFlag{
 			Name:    "repo-owner",
 			Usage:   "owner of the current repo",
-			EnvVars: []string{"CI_REPO_OWNER", "DRONE_REPO_OWNER"},
+			Sources: cli.EnvVars("CI_REPO_OWNER", "DRONE_REPO_OWNER"),
 		},
 		&cli.StringFlag{
 			Name:    "repo-name",
 			Usage:   "name of the current repo",
-			EnvVars: []string{"CI_REPO_NAME", "DRONE_REPO_NAME"},
+			Sources: cli.EnvVars("CI_REPO_NAME", "DRONE_REPO_NAME"),
 		},
 		&cli.StringFlag{
 			Name:    "pull-request-id",
 			Usage:   "id of the current pull-request",
-			EnvVars: []string{"CI_COMMIT_PULL_REQUEST", "DRONE_PULL_REQUEST"},
+			Sources: cli.EnvVars("CI_COMMIT_PULL_REQUEST", "DRONE_PULL_REQUEST"),
 		},
 		&cli.StringFlag{
 			Name:    "forge-type",
 			Usage:   "type of the forge",
-			EnvVars: []string{"CI_FORGE_TYPE", "PLUGIN_FORGE_TYPE"},
+			Sources: cli.EnvVars("CI_FORGE_TYPE", "PLUGIN_FORGE_TYPE"),
 		},
 		&cli.StringFlag{
 			Name:    "forge-url",
 			Usage:   "base url of the forge",
-			EnvVars: []string{"CI_FORGE_URL", "PLUGIN_FORGE_URL"},
+			Sources: cli.EnvVars("CI_FORGE_URL", "PLUGIN_FORGE_URL"),
 		},
 		&cli.StringFlag{
 			Name:    "forge-repo-token",
 			Usage:   "type of the forge",
-			EnvVars: []string{"CI_FORGE_REPO_TOKEN", "PLUGIN_FORGE_REPO_TOKEN"},
+			Sources: cli.EnvVars("CI_FORGE_REPO_TOKEN", "PLUGIN_FORGE_REPO_TOKEN"),
 		},
 	}
 
-	if err := app.Run(os.Args); err != nil {
+	ctx := context.Background()
+	if err := app.Run(ctx, os.Args); err != nil {
```

Comment:
Not re-used, so pass it directly I would say. 

```suggestion
	if err := app.Run(context.Background(), os.Args); err != nil {
```